packages/flutter/lib/src/isolate/isolate_logger.dart

```diff
@@ -0,0 +1,81 @@
+import 'dart:developer' as developer;
+
+import 'package:meta/meta.dart';
+
+import '../../sentry_flutter.dart';
+
+/// Static logger for Isolates that writes diagnostic messages to `dart:developer.log`.
+///
+/// Intended for worker/background isolates where a `SentryOptions` instance
+/// or hub may not be available. Because Dart statics are isolate-local,
+/// you must call [configure] once per isolate before using [log].
+class IsolateLogger {
+  IsolateLogger._();
+
+  static late bool _debug;
+  static late SentryLevel _level;
+  static late String _loggerName;
+  static bool _isConfigured = false;
+
+  /// Configures this logger for the current isolate.
+  ///
+  /// Must be called once per isolate before invoking [log].
+  /// Throws [StateError] if called more than once without calling [reset] first.
+  ///
+  /// - [debug]: when false, suppresses all logs except [SentryLevel.fatal].
+  /// - [level]: minimum severity threshold (inclusive) when [debug] is true.
+  /// - [loggerName]: logger name for the call sites
+  static void configure(
+      {required bool debug,
+      required SentryLevel level,
+      required String loggerName}) {
+    if (_isConfigured) {
+      throw StateError(
+          'IsolateLogger.configure has already been called. It can only be configured once per isolate.');
+    }
```

Comment:
We gotta use a static logger since we cannot pass `SentryFlutterOptions` to another isolate as it's not serializable